### PR TITLE
Added smoke test for log in, search, add to cart and checkout process

### PR DESCRIPTION
**Pull request by MakGlavocevic**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,45 @@
-# automation-protractor
\ No newline at end of file
+usage: git [--version] [--help] [-C <path>] [-c <name>=<value>]
+           [--exec-path[=<path>]] [--html-path] [--man-path] [--info-path]
```
Comment (MCicic) [inline README.md]: Can you remove all ununnecessary text here ?
Read me should contain instructions for running test.
List all commands user needs to do in order to run smoke test (include installation also, for example instal node js version x etc)

File: test/smoke_spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+var PageObject = require('../page_objects/page_object_model')
```
Comment (MCicic) [inline test/smoke_spec.js]: Please add ; at the end (this repeats through code)

File: test/smoke_spec.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+describe('smoke test demo', function() {
```
Comment (MCicic) [inline test/smoke_spec.js]: Here add smoke test ID also, for example describe(' 01: Smoke test', function() { .. "

File: test/smoke_spec.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+var pageObject
+
+beforeEach(function(){
+
+  pageObject = new PageObject()
+
+  browser.waitForAngularEnabled(false);
+
+  browser.get('http://automationpractice.com/index.php');
```
Comment (MCicic) [inline test/smoke_spec.js]: Can you make one separated folder named "Data" with one data.js class where you will store every data you use (every link, input, text etc).
Reason: If at some point data changes, you do not need to change your test, only data class.

File: test/smoke_spec.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+})
+
+
+      it('shall not smoke', function() {
```
Comment (MCicic) [inline test/smoke_spec.js]: Test suite title and ID should match text inside "describe", and test case title and ID should match text inside "it".
Then, you will be in sync with documentation you wrote in docs.

For example, it should be 
"it('ST-001: Sign in with valid credentials, add item to shopping cart and successfully check out', function() { .."

File: test/smoke_spec.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  
+      expect(browser.getTitle()).toEqual('My Store');
+
+      pageObject.signinButton.click();
+      
+      pageObject.emailField.sendKeys('abhtestinginternship@gmail.com');
+      pageObject.passwordField.sendKeys('!Qwerty123');
```
Comment (MCicic) [inline test/smoke_spec.js]: Can you investigate is there way to keep password out of the code, and to put it as parameter in console run command ?

File: page_objects/page_object_model.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
+var PageObject = function () {}
```
Comment (MCicic) [inline page_objects/page_object_model.js]: Can you try to refactor this class ?

Idea of POM is to have separated class for every page, for example for log in page you can have something like this:

class LoginPage  {

//GETTERS
//here you put all locators for log in, for example email fields, password field, login button etc


//ACTIONS
//Here you put basic action related to getters above.
For example method "enterEmail", "enterPassword", "ClickLogIn", "DeletePassword", "ClickCancel". etc
}

And then, inside your Smoke test you just make call to these methods.
Something like this inside Smoke test:

LoginPage.enterEmail(data.email)
                  .enterPassword(data. password)
                  .clickLogIn().. etc

Comment (MCicic) [reply page_objects/page_object_model.js]: https://www.toolsqa.com/selenium-webdriver/page-object-model/
useful link to see this concept

File: test/smoke_spec.js (hunk 6 above)
Comment (MakGlavocevic) [reply test/smoke_spec.js]: I can try perhaps with environment variables

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (MCicic) [inline .gitignore]: add node modules to git ignore (node modules should not be pushed to master later)

**Review by MCicic — APPROVED** [review]

**Pull request merged**